src/components/common-components/button-component/button.component.tsx

```diff
@@ -0,0 +1,12 @@
+import { ButtonProps } from "@material-ui/core";
+import React, { Ref } from "react";
+
+export type ButtonOptions = Omit<ButtonProps, "variant"> & {
+  variant?: "primary" | "secondary" | "outlined";
+};
+
+export const Button = React.forwardRef(
```

Comment:
To wygląda tak, ponieważ w Reakcie nie da się inaczej przekazać referencji (z tego co rozumiem, referencje - zmienne do których react może przypsiać element z DOM). Potrzebowałem tego, do `dropdowna`

https://reactjs.org/docs/forwarding-refs.html